src/components/LeafletMap.svelte

```diff
@@ -0,0 +1,126 @@
+<script>
+  import deepEqual from 'deep-equal';
+  import throttle from 'lodash.throttle';
+  import { createEventDispatcher, onMount, onDestroy } from 'svelte';
+  import * as L from 'leaflet';
+  import 'leaflet/dist/leaflet.css';
+
+  export let index;
+  export let id;
+  export let center;
+  export let zoom;
+  export let mapStyle;
+  export let numberOfMaps;
+
+  const dispatch = createEventDispatcher();
+
+  let style = {};
+  let url;
+
+  let map;
+  let layer;
+  let mapViewProps = {};
+
+  // We can set style (an object) here because mapStyle only changes when it needs to
+  $: ({ style, url } = mapStyle);
+
+  // We group map-view props here as they are useful in a few contexts
+  $: mapViewProps = { center, zoom };
+
+  const getCurrentMapView = () => {
+    return {
+      center: map.getCenter(),
+      zoom: map.getZoom() - 1,
```

Comment:
I still question whether this is the desired user experience since they may want to compare actual zoom level to the same actual zoom level across map types. That said, since this is already the pattern for Google maps as well, maybe we can open a followup issue to make this a choice for the user if they're using a map that won't be aligned?

Reply:
Sure let's address this separately.